### PR TITLE
continue CSS Consistency

### PR DESCRIPTION
**Pull request by joshgoebel**
### Changes

Continues work towards CSS consistency across all our CSS grammars
that allow raw CSS (ie, they are supersets).  Cleans up all grammars a lot.

Less is still a hot mess, but the other 3 are getting a lot cleaner.

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`
- [x] Added myself to `AUTHORS.txt`, under Contributors

### CODE REVIEW

**Comment by joshgoebel** [comment]
Should be easy to review this one one commit at a time if the overall view is too messy.  I tried to keep them clean.

**Review by sa-orlov — APPROVED** [review]

**Review by sa-orlov — APPROVED** [review]

File: src/languages/lib/css-shared.js [hunk 1]
```diff
@@ -1,3 +1,137 @@
+export const MODES = (hljs) => {
```
Comment (egor-rogov) [inline src/languages/lib/css-shared.js:1]: Doesn't `MODES` sound too broad for these 3 modes? You'll have to guess what `  const modes = css.MODES(hljs);` really means.

File: src/languages/css.js [hunk 2]
```diff
@@ -5,141 +5,144 @@ Website: https://developer.mozilla.org/en-US/docs/Web/CSS
...
-          hljs.APOS_STRING_MODE,
-          hljs.QUOTE_STRING_MODE,
+        className: 'selector-pseudo',
+        variants: [
+          {
+            begin: ':(' + css.PSEUDO_CLASSES.join('|') + ')'
+          },
+          {
+            begin: '::(' + css.PSEUDO_ELEMENTS.join('|') + ')'
+          }
```
Comment (egor-rogov) [inline src/languages/css.js:66]: Are two variants better that one with `::?`?

Comment (joshgoebel) [reply src/languages/css.js:66]: CLASSES and ELEMENTS are entirely separate lists.  This is more precise (and clearer).

File: src/languages/lib/css-shared.js (hunk 1 above)
Comment (joshgoebel) [reply src/languages/lib/css-shared.js:1]: Seems clear to me?   MODES just for CSS vs MODES in general.

Not sure I follow.  Did you have an alternative suggestion that you think is clearer?

I wrapped them so I could pass in `hljs`... seems silly to have a 3+ separate functions and have to pass them all hljs.

Comment (egor-rogov) [reply src/languages/lib/css-shared.js:1]: No, I don't have anything else but 3 separate functions. OK if single function seems clear enough to you.

**Review by egor-rogov — APPROVED** [review]
Looks good to me.

**Pull request merged**